### PR TITLE
Implement Database’s performMaintenance(type:)

### PR DESCRIPTION
**Pull request by pasin**
* Implemented Database’s performMaintenance(type:)
* Deprecate compact method.

CBL-991

### CODE REVIEW

File: Objective-C/CBLDatabase.h [hunk 1]
```diff
@@ -244,14 +253,23 @@ typedef NS_ENUM(uint32_t, CBLConcurrencyControl) {
...
+ Performs database maintenance.
+ 
+ @param type Maintenance type.
+ @param error On return, the error if any.
+ @return True on success, false on failure.
+ */
+- (BOOL) performMaintenance: (CBLMaintenanceType)type error: (NSError**)error;
+
 /**
  Compacts the database file by deleting unused attachment files and vacuuming
```
Comment (jayahariv) [inline Objective-C/CBLDatabase.h:266]: " This method is deprecated. " text in the description?

Comment (pasin) [reply Objective-C/CBLDatabase.h:266]: Yeah, will add that.

File: Swift/Tests/DatabaseTest.swift [hunk 2]
```diff
@@ -103,6 +103,181 @@ class DatabaseTest: CBLTestCase {
...
+        for doc in docs {
+            let doc = db.document(withID: doc.id)!
+            try db.deleteDocument(doc)
+            XCTAssertNil(db.document(withID: doc.id))
+        }
+        XCTAssertEqual(db.count, 0)
+        
+        // Integrity Check:
+        try db.performMaintenance(type: .integrityCheck)
+    }
```
Comment (jayahariv) [inline Swift/Tests/DatabaseTest.swift:280]: is it possible to test a corrupted database?

File: Swift/Database.swift [hunk 3]
```diff
@@ -309,10 +320,18 @@ public final class Database {
...
     }
     
+    /// Performs database maintenance.
+    ///
+    /// - Throws: An error on a failure
+    public func performMaintenance(type: MaintenanceType) throws {
+        try _impl.perform(CBLMaintenanceType(rawValue: UInt32(type.rawValue))!)
+    }
+    
     /// Compacts the database file by deleting unused attachment files and
```
Comment (jayahariv) [inline Swift/Database.swift:330]: same

File: Swift/Tests/DatabaseTest.swift (hunk 2 above)
Comment (pasin) [reply Swift/Tests/DatabaseTest.swift:280]: I don't have the corrupted db and I don't know how to produce one either.

Comment (jayahariv) [reply Swift/Tests/DatabaseTest.swift:280]: https://issues.couchbase.com/browse/CBL-1046

File: Objective-C/CBLDatabase.h (hunk 1 above)
Comment (pasin) [reply Objective-C/CBLDatabase.h]: Done

File: Swift/Database.swift (hunk 3 above)
Comment (pasin) [reply Swift/Database.swift]: Done

**Pull request merged**